<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction | Applied Statistical Methods in Animal Sciences</title>
  <meta name="description" content="Chapter 1 Introduction | Applied Statistical Methods in Animal Sciences" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction | Applied Statistical Methods in Animal Sciences" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/charlotte-ngs/gelasmss2021" />
  
  
  <meta name="github-repo" content="charlotte-ngs/gelasmss2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction | Applied Statistical Methods in Animal Sciences" />
  
  
  

<meta name="author" content="Peter von Rohr" />


<meta name="date" content="2021-02-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="intro-linalg.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>



<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Applied Statistical Methods</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#general-developments"><i class="fa fa-check"></i>General Developments</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#where-does-this-course-fit-in"><i class="fa fa-check"></i>Where Does This Course Fit In?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-objectives"><i class="fa fa-check"></i>Course Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i>Prerequisites</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="asm-intro.html"><a href="asm-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="asm-intro.html"><a href="asm-intro.html#us-presidential-campaigns"><i class="fa fa-check"></i><b>1.1</b> US Presidential Campaigns</a></li>
<li class="chapter" data-level="1.2" data-path="asm-intro.html"><a href="asm-intro.html#health-care"><i class="fa fa-check"></i><b>1.2</b> Health Care</a></li>
<li class="chapter" data-level="1.3" data-path="asm-intro.html"><a href="asm-intro.html#face-recognition"><i class="fa fa-check"></i><b>1.3</b> Face Recognition</a></li>
<li class="chapter" data-level="1.4" data-path="asm-intro.html"><a href="asm-intro.html#feed-intake-and-behavior-traits-of-cows"><i class="fa fa-check"></i><b>1.4</b> Feed Intake and Behavior Traits of Cows</a></li>
<li class="chapter" data-level="1.5" data-path="asm-intro.html"><a href="asm-intro.html#conclusions-from-examples"><i class="fa fa-check"></i><b>1.5</b> Conclusions from Examples</a></li>
<li class="chapter" data-level="1.6" data-path="asm-intro.html"><a href="asm-intro.html#asm-traditional-animal-breeding"><i class="fa fa-check"></i><b>1.6</b> Traditional Livestock Breeding</a></li>
<li class="chapter" data-level="1.7" data-path="asm-intro.html"><a href="asm-intro.html#asm-genomic-selection"><i class="fa fa-check"></i><b>1.7</b> Genomic Selection</a></li>
<li class="chapter" data-level="1.8" data-path="asm-intro.html"><a href="asm-intro.html#asm-mono-genic-model"><i class="fa fa-check"></i><b>1.8</b> Mono-Genic Model</a></li>
<li class="chapter" data-level="1.9" data-path="asm-intro.html"><a href="asm-intro.html#asm-two-step-approach"><i class="fa fa-check"></i><b>1.9</b> Two Step Approach</a></li>
<li class="chapter" data-level="1.10" data-path="asm-intro.html"><a href="asm-intro.html#asm-single-step-approach"><i class="fa fa-check"></i><b>1.10</b> Single Step Approach</a></li>
<li class="chapter" data-level="1.11" data-path="asm-intro.html"><a href="asm-intro.html#asm-summary"><i class="fa fa-check"></i><b>1.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro-linalg.html"><a href="intro-linalg.html"><i class="fa fa-check"></i><b>2</b> Introduction To Linear Algebra</a><ul>
<li class="chapter" data-level="2.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-glimpse-ahead"><i class="fa fa-check"></i><b>2.1</b> Glimpse Ahead</a></li>
<li class="chapter" data-level="2.2" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-vectors"><i class="fa fa-check"></i><b>2.2</b> Vectors</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-physics-perspective"><i class="fa fa-check"></i><b>2.2.1</b> Physics Perspective</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-computer-science-perspective"><i class="fa fa-check"></i><b>2.2.2</b> Computer Science Perspective</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-geometric-context"><i class="fa fa-check"></i><b>2.2.3</b> Geometric Context</a></li>
<li class="chapter" data-level="2.2.4" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-coordinate-system"><i class="fa fa-check"></i><b>2.2.4</b> Coordinate System</a></li>
<li class="chapter" data-level="2.2.5" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-vector-operations"><i class="fa fa-check"></i><b>2.2.5</b> Vector Operations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-matrices"><i class="fa fa-check"></i><b>2.3</b> Matrices</a><ul>
<li class="chapter" data-level="2.3.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-special-matrices"><i class="fa fa-check"></i><b>2.3.1</b> Special Matrices</a></li>
<li class="chapter" data-level="2.3.2" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-matrix-operation"><i class="fa fa-check"></i><b>2.3.2</b> Matrix Operations</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-systems-of-equations"><i class="fa fa-check"></i><b>2.4</b> Systems Of Equations</a><ul>
<li class="chapter" data-level="2.4.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-matrix-vector-notation"><i class="fa fa-check"></i><b>2.4.1</b> Matrix-Vector Notation</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-solving-systems-of-linear-equations"><i class="fa fa-check"></i><b>2.5</b> Solving Systems of Linear Equations</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quan-gen.html"><a href="quan-gen.html"><i class="fa fa-check"></i><b>3</b> Basics in Quantitative Genetics</a><ul>
<li class="chapter" data-level="3.1" data-path="quan-gen.html"><a href="quan-gen.html#single-locus-quant-trait"><i class="fa fa-check"></i><b>3.1</b> Single Locus - Quantitative Trait</a><ul>
<li class="chapter" data-level="3.1.1" data-path="quan-gen.html"><a href="quan-gen.html#qg-terminology"><i class="fa fa-check"></i><b>3.1.1</b> Terminology</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quan-gen.html"><a href="quan-gen.html#qg-frequency"><i class="fa fa-check"></i><b>3.2</b> Frequencies</a></li>
<li class="chapter" data-level="3.3" data-path="quan-gen.html"><a href="quan-gen.html#hw-eq"><i class="fa fa-check"></i><b>3.3</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="3.4" data-path="quan-gen.html"><a href="quan-gen.html#value-mean"><i class="fa fa-check"></i><b>3.4</b> Value and Mean</a><ul>
<li class="chapter" data-level="3.4.1" data-path="quan-gen.html"><a href="quan-gen.html#geno-value"><i class="fa fa-check"></i><b>3.4.1</b> Genotypic Values</a></li>
<li class="chapter" data-level="3.4.2" data-path="quan-gen.html"><a href="quan-gen.html#pop-mean"><i class="fa fa-check"></i><b>3.4.2</b> Population Mean</a></li>
<li class="chapter" data-level="3.4.3" data-path="quan-gen.html"><a href="quan-gen.html#breed-value"><i class="fa fa-check"></i><b>3.4.3</b> Breeding Values</a></li>
<li class="chapter" data-level="3.4.4" data-path="quan-gen.html"><a href="quan-gen.html#allele-substitution"><i class="fa fa-check"></i><b>3.4.4</b> Allele Substitution</a></li>
<li class="chapter" data-level="3.4.5" data-path="quan-gen.html"><a href="quan-gen.html#dominance-deviation"><i class="fa fa-check"></i><b>3.4.5</b> Dominance Deviation</a></li>
<li class="chapter" data-level="3.4.6" data-path="quan-gen.html"><a href="quan-gen.html#summary-of-values"><i class="fa fa-check"></i><b>3.4.6</b> Summary of Values</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="quan-gen.html"><a href="quan-gen.html#variances"><i class="fa fa-check"></i><b>3.5</b> Variances</a></li>
<li class="chapter" data-level="3.6" data-path="quan-gen.html"><a href="quan-gen.html#extension-to-more-loci"><i class="fa fa-check"></i><b>3.6</b> Extension To More Loci</a><ul>
<li class="chapter" data-level="3.6.1" data-path="quan-gen.html"><a href="quan-gen.html#epistatic-interaction"><i class="fa fa-check"></i><b>3.6.1</b> Epistatic Interaction</a></li>
<li class="chapter" data-level="3.6.2" data-path="quan-gen.html"><a href="quan-gen.html#interaction-variance"><i class="fa fa-check"></i><b>3.6.2</b> Interaction Variance</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="quan-gen.html"><a href="quan-gen.html#genetic-models"><i class="fa fa-check"></i><b>3.7</b> Genetic Models</a><ul>
<li class="chapter" data-level="3.7.1" data-path="quan-gen.html"><a href="quan-gen.html#model-usage-in-routine-evaluations"><i class="fa fa-check"></i><b>3.7.1</b> Model Usage In Routine Evaluations</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="quan-gen.html"><a href="quan-gen.html#appendix-derivations"><i class="fa fa-check"></i><b>3.8</b> Appendix: Derivations</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Statistical Methods in Animal Sciences</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="asm-intro" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>According to Wikipedia <span class="citation">(Wikipedia <a href="#ref-Wikipedia2019" role="doc-biblioref">2019</a>)</span>, the term <code>Big Data</code> has been used since the 1990s. Some credit was given to John Mashey <span class="citation">(Mashey <a href="#ref-Mashey1998" role="doc-biblioref">1998</a>)</span> for popularizing the term. Nowadays <code>Big Data</code> is used in connection with large companies, social media or governments which collect massive amounts of data. This data is then used to infer certain conclusions about behaviors of customers, or followers or voters. The following subsections show a few examples of <code>Big Data</code>-applications.</p>
<div id="us-presidential-campaigns" class="section level2">
<h2><span class="header-section-number">1.1</span> US Presidential Campaigns</h2>
<p>The presidential election campaigns of Barack Obama were examples of how <code>Big Data</code> was used to access behaviors of voters <span class="citation">(Issenberg <a href="#ref-Issenberg2013" role="doc-biblioref">2013</a>)</span>.</p>
</div>
<div id="health-care" class="section level2">
<h2><span class="header-section-number">1.2</span> Health Care</h2>
<p>A different example is the use of <code>Big Data</code> in health care. An overview of the use of <code>Big Data</code> in health care is given in <span class="citation">(Adibuzzaman et al. <a href="#ref-Adibuzzaman2017" role="doc-biblioref">2017</a>)</span>. The collected health data is most likely not only used by research but also by insurance companies.</p>
</div>
<div id="face-recognition" class="section level2">
<h2><span class="header-section-number">1.3</span> Face Recognition</h2>
<p>The Swiss TV news show <code>10 vor 10</code> showed on the <span class="math inline">\(7^{th}\)</span> Feb. 2020 how a data journalist managed to build a face recognition system. The general idea how this system works is shown in Figure <a href="asm-intro.html#fig:facerecog">1.1</a>.</p>
<div class="figure"><span id="fig:facerecog"></span>
<img src="odg/facerecog.png" alt="Design of Face Recognition System" width="11cm" />
<p class="caption">
Figure 1.1: Design of Face Recognition System
</p>
</div>
<p>The main goal of the face recognition system was to be able to identify certain persons, in this case the politicians that had a picture on the platform ‘smartvote’, on random pictures obtained from the social media platform Instagram. The data used for the face recognition system can be split into two parts.</p>
<ol style="list-style-type: decimal">
<li><strong>Training</strong> set. The training set consists of pictures showing different politicians. This dataset was downloaded from the politics platform <a href="https://smartvote.ch/">smartvote</a>. The training set is used to establish a fixed relation between pictures of politicians and their names. Furthermore the pictures of the politicians are also used to extract characteristic features that are different among the different pictures of the different politicians. Most of these features consist of numbers which describe the faces shown on the pictures. Examples of such features might be the surface of the face, the surface of the hairs shown in the picture, the length of the mouth, etc.<br />
</li>
<li><strong>Test</strong> set: The test set consists of <span class="math inline">\(230000\)</span> publicly available pictures on the platform Instagram. The content of these pictures is a priori unknown. The question that the face recognition system tries to answer is whether it is possible to identify any of the politicians from the training set on any of the Instagram-pictures. This question is answered by a comparison of the features extracted from the instagram pictures to the features that were obtained from the pictures of the training set. If the feature comparison results in a match, the system suggests that we found a given person on one of the instagram pictures.</li>
</ol>
<p>The complete story about the face recognition system is available under <a href="https://www.srf.ch/news/schweiz/automatische-gesichtserkennung-so-einfach-ist-es-eine-ueberwachungsmaschine-zu-bauen" class="uri">https://www.srf.ch/news/schweiz/automatische-gesichtserkennung-so-einfach-ist-es-eine-ueberwachungsmaschine-zu-bauen</a>.</p>
</div>
<div id="feed-intake-and-behavior-traits-of-cows" class="section level2">
<h2><span class="header-section-number">1.4</span> Feed Intake and Behavior Traits of Cows</h2>
<p>In the recent past technologies based on computer vision have been introduced into agricultural applications. Two examples of such applications are</p>
<ol style="list-style-type: decimal">
<li>Estimation of feed intake of cows based on video data as described by <span class="citation">(Chizzotti et al. <a href="#ref-Chizzotti2015" role="doc-biblioref">2015</a>)</span>.</li>
<li>An ongoing EU-Interreg project called “<a href="https://www.alpine-space.eu/projects/sesam/en/home">SESAM</a>” aims at predicting basic behavior traits from data obtained from sensors an from video recordings.</li>
</ol>
</div>
<div id="conclusions-from-examples" class="section level2">
<h2><span class="header-section-number">1.5</span> Conclusions from Examples</h2>
<p>The above shown examples demonstrate that data can be used for very different purposes. Using just one source of data does in most cases not give a lot of insights. But when different sources of information are combined, they can be used to make certain predictions that influences our daily lives. Hence this kind of development is becoming a general interest to all of us. In what follows, we try to show that some of these methods have been applied for a long time in the area of animal science and especially in livestock breeding.</p>
</div>
<div id="asm-traditional-animal-breeding" class="section level2">
<h2><span class="header-section-number">1.6</span> Traditional Livestock Breeding</h2>
<p>In livestock breeding the statistical analyses that are used together with <code>Big Data</code> technologies have long been applied to predict breeding values for livestock populations. The process of breeding value prediction uses statistical methods to assess the genetic potential of breeding animals in a population. The data used to predict the breeding values are collected mainly for quality control or management purposes. The prediction of breeding values can be viewed as a side product. In the area of cattle breeding, data collection consists of rather complex flows of information. The flow of information is shown in Figure <a href="asm-intro.html#fig:datacollectionflow">1.2</a>.</p>
<div class="figure"><span id="fig:datacollectionflow"></span>
<img src="odg/datacollectionflow.png" alt="Data Flow in an Animal Breeding Program" width="11cm" />
<p class="caption">
Figure 1.2: Data Flow in an Animal Breeding Program
</p>
</div>
</div>
<div id="asm-genomic-selection" class="section level2">
<h2><span class="header-section-number">1.7</span> Genomic Selection</h2>
<p>The data flow shown in Figure <a href="asm-intro.html#fig:datacollectionflow">1.2</a> contains the traditional evaluation of data to result in predicted breeding values. But it is missing the newest development in the breeding industry. This development is known as <code>Genomic Selection</code> (GS). GS was introduced by the work of <span class="citation">(Meuwissen, Hayes, and Goddard <a href="#ref-Meuwissen2001" role="doc-biblioref">2001</a>)</span>. The methods presented by <span class="citation">(Meuwissen, Hayes, and Goddard <a href="#ref-Meuwissen2001" role="doc-biblioref">2001</a>)</span> were only introduced into practical breeding programs when <span class="citation">(Schaeffer <a href="#ref-Schaeffer2006" role="doc-biblioref">2006</a>)</span> showed the tremendous potential of saving costs for breeding programs. The use of <strong>genomic</strong> information for the assessment of the genetic potential of all breeding animals represents the core of the evaluation approach presented by <span class="citation">(Meuwissen, Hayes, and Goddard <a href="#ref-Meuwissen2001" role="doc-biblioref">2001</a>)</span>. The term <code>genomic</code> is used because genetic markers which are evenly spaced over the complete genome are used as information source. Single Nucleotide Polymorphisms (SNP) are the most widely used marker model nowadays. SNPs are single positions in the genome that occur in different variants in the whole population. A description on how to identify SNPs in a population is given in <span class="citation">(Czech et al. <a href="#ref-Czech2018" role="doc-biblioref">2018</a>)</span>. Potential use cases of SNPs are outlined by <span class="citation">(Seidel, Jr. <a href="#ref-SeidelJr.2010" role="doc-biblioref">2010</a>)</span> and <span class="citation">(Pant et al. <a href="#ref-Pant2012" role="doc-biblioref">2012</a>)</span>. The genetic configuration of an SNP in a given population is shown in Figure <a href="asm-intro.html#fig:snpgeneticconfiguration">1.3</a>.</p>
<div class="figure"><span id="fig:snpgeneticconfiguration"></span>
<img src="odg/snpgeneticconfiguration.png" alt="Genetic Configuration of a Single Nucleotide Polymorphism (SNP)" width="440" />
<p class="caption">
Figure 1.3: Genetic Configuration of a Single Nucleotide Polymorphism (SNP)
</p>
</div>
<p>These SNPs can occur anywhere in the genome which means they can be observed in coding regions, in non-coding regions as well as in regulatory regions. In genomic selection, we are working with a large set of SNPs that are distributed over the complete genome. Hence some of the SNPs will be located close to genetic positions that are important for the expression of quantitative traits of interest. Such genetic positions which are related to quantitative traits are often called <code>Quantitative Trait Loci</code> (QTL). QTL themselves are difficult to detect and their inheritance is often manifested in complex modes. But due to the likely occurrence of several SNPs in the close proximity of a QTL, the inheritance of QTL alleles and of surrounding SNP alleles will not be independent due to linkage between SNPs and QTL. Such a linkage scenario between two SNPs flanking a QTL is shown in Figure <a href="asm-intro.html#fig:linkagesnpqtl">1.4</a>.</p>
<div class="figure"><span id="fig:linkagesnpqtl"></span>
<img src="odg/linkagesnpqtl.png" alt="Two SNPs flanking a QTL" width="440" />
<p class="caption">
Figure 1.4: Two SNPs flanking a QTL
</p>
</div>
<p>Although the QTL is likely to span a range of many positions on the chromosome, we can still assume the QTL to be bi-allelic with alleles <span class="math inline">\(Q_1\)</span> and <span class="math inline">\(Q_2\)</span>. In theory, any SNP position can have four different alleles according to the four different bases. But when looking at different SNPs in real-world populations, most of them only show two alleles. Hence, for the two SNPs flanking the QTL shown in Figure <a href="asm-intro.html#fig:linkagesnpqtl">1.4</a> they also have just two alleles <span class="math inline">\(SNP1_1\)</span>, <span class="math inline">\(SNP1_2\)</span>, <span class="math inline">\(SNP2_1\)</span> and <span class="math inline">\(SNP2_2\)</span>. In genetics the dependency of the inheritance of neighboring loci (marker or QTL) is referred to as <code>linkage disequilibrium</code> (LD). This means that any joint allele frequency <span class="math inline">\(Pr(SNP1_i, Q_j, SNP2_k)\)</span> does not correspond to the product of the single allele frequencies of the two SNPs (<span class="math inline">\(SNP1\)</span> and <span class="math inline">\(SNP2\)</span>) and the QTL. In a formula this can be written as</p>
<p><span class="math display" id="eq:allelfreqsnpqtllinkage">\[\begin{equation}
 Pr(SNP1_i, Q_j, SNP2_k) \ne Pr(SNP1_i) * Pr(Q_j) * Pr(SNP2_k)
 \tag{1.1}
\end{equation}\]</span></p>
<p>Assuming that the QTL allele <span class="math inline">\(Q_1\)</span> is favorable for the expression of a given trait of interest and using the fact of LD as expressed in <a href="asm-intro.html#eq:allelfreqsnpqtllinkage">(1.1)</a>, the alleles of <span class="math inline">\(SNP1\)</span> and <span class="math inline">\(SNP2\)</span> which occur more frequently together with <span class="math inline">\(Q_1\)</span> are therefore also related to favorable expression levels of the trait of interest. In real breeding populations, the position of the QTL is unknown. But because we know the allelic configuration of a large number of SNP loci from many breeding animals, we can reliably relate SNP alleles and favorable expression levels of traits of interest.</p>
</div>
<div id="asm-mono-genic-model" class="section level2">
<h2><span class="header-section-number">1.8</span> Mono-Genic Model</h2>
<p>In quantitative genetics, the so-called mono-genic or single-locus model allows us to quantify the genetic potential of breeding animals in terms of breeding values. The standard reference in quantitative genetics in which also the mono-genic model is described is <span class="citation">(Falconer and Mackay <a href="#ref-Falconer1996" role="doc-biblioref">1996</a>)</span>. For a single locus, the breeding value depends on the allele frequencies at that locus and on the additive substitution effect which is often called <span class="math inline">\(\alpha\)</span>. The mono-genic model for any given SNP locus in relation to the level of expression of a given trait of interest can be visualized in the following Figure <a href="asm-intro.html#fig:monogenicsnpmodel">1.5</a>.</p>
<div class="figure"><span id="fig:monogenicsnpmodel"></span>
<img src="odg/monogenicsnpmodel.png" alt="Single-Locus Model for a Quantitative Trait" width="11cm" />
<p class="caption">
Figure 1.5: Single-Locus Model for a Quantitative Trait
</p>
</div>
<p>In a real breeding population, we assume that the effect of all loci linked to the SNPs are purely additive. Hence any values for <span class="math inline">\(d\)</span> are all zero. As a consequence of that the breeding values at any given SNP position only depend on the allele frequencies of the SNP and the <span class="math inline">\(a\)</span> values at every SNP. The overall breeding value of a given animal is computed as the sum of all locus-specific breeding values. This overall breeding value is called <strong>genomic breeding value</strong> (GBV). In order to get an estimate of such a GBV, we have to estimate all <span class="math inline">\(a\)</span> values at any SNP position. This estimation procedure can be done in one of the following two ways.</p>
<ol style="list-style-type: decimal">
<li>Two step approach</li>
<li>Single step approach</li>
</ol>
</div>
<div id="asm-two-step-approach" class="section level2">
<h2><span class="header-section-number">1.9</span> Two Step Approach</h2>
<p>In the two step approach the estimation of the <span class="math inline">\(a\)</span>-values and the computation of the GBVs are done in two separate steps. For the estimation of the <span class="math inline">\(a\)</span> values for all SNPs, a reference population is defined. In dairy cattle breeding this reference population consists of all male breeding animals. In the recent past, the reference population has been augmented continuously with female animals. The animals in the reference population are all genotyped and they also all have phenotypic measurements<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> for the trait of interest. The estimation of the <span class="math inline">\(a\)</span> values amounts to estimating fixed effects in a linear model. We will see in later chapters of this course what methods are available to estimate these parameters.</p>
<p>In the second step the estimates for all the <span class="math inline">\(a\)</span> values are used to compute the GBVs for all animals with genomic SNP information also for those outside of the reference population. The Figure <a href="asm-intro.html#fig:twostepgs">1.6</a> tries to summarize the process graphically.</p>
<div class="figure"><span id="fig:twostepgs"></span>
<img src="odg/twostepgs.png" alt="Two Step Approach To Estimate Genomic Breeding Values" width="11cm" />
<p class="caption">
Figure 1.6: Two Step Approach To Estimate Genomic Breeding Values
</p>
</div>
<p>The big advantage of the two step method is that once we have defined a good reference population which yields reliable estimates for the <span class="math inline">\(a\)</span> values, the computation of the GBV is a simple computation of just summing up the <span class="math inline">\(a\)</span> contributions with the correct sign determined by the SNP genotypes of the animals for which the GBVs should be determined. All animals with SNP genotypes can get GBV values. The difficult part in the two step approach is to define a reliable reference population and to determine good phenotypic measurements (<span class="math inline">\(y\)</span>).</p>
</div>
<div id="asm-single-step-approach" class="section level2">
<h2><span class="header-section-number">1.10</span> Single Step Approach</h2>
<p>The estimation of the <span class="math inline">\(a\)</span> values and the prediction of the genomic breeding values is done in one step using linear mixed effects models. In this single step evaluation animals with and without genomic information can get predicted genomic breeding values in a single analysis. One possibility to get to this predicted breeding values is via the use of <code>Genomic BLUP</code> (GBLUP). This will be the topic of a complete chapter in this course. The problem with the single step approach is to get an estimate of the covariance between animals with and without genomic information. This is a problem of ongoing research.</p>
</div>
<div id="asm-summary" class="section level2">
<h2><span class="header-section-number">1.11</span> Summary</h2>
<p>The main difference between traditional predictions of breeding values using a BLUP animal model and the prediction of GBV is that the former uses the so called <strong>infinitesimal</strong> model to assess the genetic potential and the latter uses sufficiently dense genomic information and uses a <strong>polygenic</strong> model. This difference is illustrated in Figure <a href="asm-intro.html#fig:infinitesimalvspolygenic">1.7</a>.</p>
<div class="figure"><span id="fig:infinitesimalvspolygenic"></span>
<img src="odg/infinitesimalvspolygenic.png" alt="Infinitesimal Versus Polygenic Model" width="11cm" />
<p class="caption">
Figure 1.7: Infinitesimal Versus Polygenic Model
</p>
</div>
<p>In the remaining chapters, different approaches for the prediction of GBVs are described. Chapter 2 gives a description of the fixed linear effects model and how it was tried to be used for GBV prediction by <span class="citation">(Meuwissen, Hayes, and Goddard <a href="#ref-Meuwissen2001" role="doc-biblioref">2001</a>)</span>. Chapter 3 introduces BLUP methodology in the context of predicting GBVs. In Chapter 4 the method called LASSO is introduced. Interestingly enough, this method is used very seldom in the area of animal breeding. Last but not least, Chapter 5 makes an excursion into Bayesian estimation approaches. The Bayesian methods are important because they are used in practical breeding programs of Swiss Dairy cattle.</p>



</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Adibuzzaman2017">
<p>Adibuzzaman, M, P DeLaurentis, J Hill, and B D Benneyworth. 2017. “Big data in healthcare - the promises, challenges and opportunities from a research perspective: A case study with a model database.” <em>AMIA Annu Symp Proc</em> 2017: 384–92. <a href="https://www.ncbi.nlm.nih.gov/pubmed/29854102">https://www.ncbi.nlm.nih.gov/pubmed/29854102</a>.</p>
</div>
<div id="ref-Chizzotti2015">
<p>Chizzotti, M. L., F. S. Machado, E. E. L. Valente, L. G. R. Pereira, M. M. Campos, T. R. Tomich, S. G. Coelho, and M. N. Ribas. 2015. “Technical note: Validation of a system for monitoring individual feeding behavior and individual feed intake in dairy cattle.” <em>Journal of Dairy Science</em> 98 (5): 3438–42. <a href="https://doi.org/10.3168/jds.2014-8925">https://doi.org/10.3168/jds.2014-8925</a>.</p>
</div>
<div id="ref-Czech2018">
<p>Czech, Bartosz, Magdalena Fraszczak, Magda Mielczarek, and Joanna Szyda. 2018. “Identification and annotation of breed-specific single nucleotide polymorphisms in Bos taurus genomes.” <em>PLoS ONE</em> 13 (6): 1–9. <a href="https://doi.org/10.1109/TCSI.2014.2341116">https://doi.org/10.1109/TCSI.2014.2341116</a>.</p>
</div>
<div id="ref-Falconer1996">
<p>Falconer, D. S., and Trudy F. C. Mackay. 1996. <em>Introduction to Quantitative Genetics</em>. 4th ed. Essex: Addison Wesley Longman Limited.</p>
</div>
<div id="ref-Issenberg2013">
<p>Issenberg, Sasha. 2013. “How Obama’s Team Used Big Data to Rally Voters.” <em>MIT Technology Review</em> 116 (1): 38–49.</p>
</div>
<div id="ref-Mashey1998">
<p>Mashey, John R. 1998. “Big Data ... and the Next Wave of InfraStress.” <a href="http://static.usenix.org/event/usenix99/invited%7B\_%7Dtalks/mashey.pdf">http://static.usenix.org/event/usenix99/invited{\_}talks/mashey.pdf</a>.</p>
</div>
<div id="ref-Meuwissen2001">
<p>Meuwissen, T H E, B J Hayes, and M E Goddard. 2001. “Prediction of Total Genetic Value Using Genome-Wide Dense Marker Maps.” <em>Genetics</em> 157: 1819–29.</p>
</div>
<div id="ref-Pant2012">
<p>Pant, Sameer D., Flavio S. Schenkel, Chris P. Verschoor, and Niel A. Karrow. 2012. “Use of breed-specific single nucleotide polymorphisms to discriminate between Holstein and Jersey dairy cattle breeds.” <em>Animal Biotechnology</em> 23 (1): 1–10. <a href="https://doi.org/10.1080/10495398.2012.636224">https://doi.org/10.1080/10495398.2012.636224</a>.</p>
</div>
<div id="ref-Schaeffer2006">
<p>Schaeffer, L. R. 2006. “Strategy for applying genome-wide selection in dairy cattle.” <em>J Anim Breed Genet</em> 123 (123): 218–23. <a href="https://doi.org/10.1111/j.1439-0388.2006.00595.x">https://doi.org/10.1111/j.1439-0388.2006.00595.x</a>.</p>
</div>
<div id="ref-SeidelJr.2010">
<p>Seidel, Jr., G. E. 2010. “Brief introduction to whole-genome selection in cattle using single nucleotide polymorphisms.” <em>Reproduction Fertility and Development</em> 22 (1): 138–44. <a href="https://doi.org/10.1071/RD09220">https://doi.org/10.1071/RD09220</a>.</p>
</div>
<div id="ref-Wikipedia2019">
<p>Wikipedia, Contributors. 2019. “Big data - Wikipedia.” <a href="https://en.wikipedia.org/w/index.php?title=Big%7B\_%7Ddata%7B\&amp;%7Doldid=881938730">https://en.wikipedia.org/w/index.php?title=Big{\_}data{\&amp;}oldid=881938730</a>.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>Whenever phenotypic measurements are not available, traditionally predicted breeding values are transformed back into pseudo phenotypes which are then used to estimate <span class="math inline">\(a\)</span> values.<a href="asm-intro.html#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="intro-linalg.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/charlotte-ngs/gelasmss2021/edit/master/cn_asm/01_intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-asmas-ss2021.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
