<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Applied Statistical Methods in Animal Sciences</title>
  <meta name="description" content="Applied Statistical Methods in Animal Sciences" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Applied Statistical Methods in Animal Sciences" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://github.com/charlotte-ngs/gelasmss2021" />
  
  
  <meta name="github-repo" content="charlotte-ngs/gelasmss2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Applied Statistical Methods in Animal Sciences" />
  
  
  

<meta name="author" content="Peter von Rohr" />


<meta name="date" content="2021-02-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="asm-intro.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>



<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Applied Statistical Methods</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#general-developments"><i class="fa fa-check"></i>General Developments</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#where-does-this-course-fit-in"><i class="fa fa-check"></i>Where Does This Course Fit In?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-objectives"><i class="fa fa-check"></i>Course Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i>Prerequisites</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="asm-intro.html"><a href="asm-intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="asm-intro.html"><a href="asm-intro.html#us-presidential-campaigns"><i class="fa fa-check"></i><b>1.1</b> US Presidential Campaigns</a></li>
<li class="chapter" data-level="1.2" data-path="asm-intro.html"><a href="asm-intro.html#health-care"><i class="fa fa-check"></i><b>1.2</b> Health Care</a></li>
<li class="chapter" data-level="1.3" data-path="asm-intro.html"><a href="asm-intro.html#face-recognition"><i class="fa fa-check"></i><b>1.3</b> Face Recognition</a></li>
<li class="chapter" data-level="1.4" data-path="asm-intro.html"><a href="asm-intro.html#feed-intake-and-behavior-traits-of-cows"><i class="fa fa-check"></i><b>1.4</b> Feed Intake and Behavior Traits of Cows</a></li>
<li class="chapter" data-level="1.5" data-path="asm-intro.html"><a href="asm-intro.html#conclusions-from-examples"><i class="fa fa-check"></i><b>1.5</b> Conclusions from Examples</a></li>
<li class="chapter" data-level="1.6" data-path="asm-intro.html"><a href="asm-intro.html#asm-traditional-animal-breeding"><i class="fa fa-check"></i><b>1.6</b> Traditional Livestock Breeding</a></li>
<li class="chapter" data-level="1.7" data-path="asm-intro.html"><a href="asm-intro.html#asm-genomic-selection"><i class="fa fa-check"></i><b>1.7</b> Genomic Selection</a></li>
<li class="chapter" data-level="1.8" data-path="asm-intro.html"><a href="asm-intro.html#asm-mono-genic-model"><i class="fa fa-check"></i><b>1.8</b> Mono-Genic Model</a></li>
<li class="chapter" data-level="1.9" data-path="asm-intro.html"><a href="asm-intro.html#asm-two-step-approach"><i class="fa fa-check"></i><b>1.9</b> Two Step Approach</a></li>
<li class="chapter" data-level="1.10" data-path="asm-intro.html"><a href="asm-intro.html#asm-single-step-approach"><i class="fa fa-check"></i><b>1.10</b> Single Step Approach</a></li>
<li class="chapter" data-level="1.11" data-path="asm-intro.html"><a href="asm-intro.html#asm-summary"><i class="fa fa-check"></i><b>1.11</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro-linalg.html"><a href="intro-linalg.html"><i class="fa fa-check"></i><b>2</b> Introduction To Linear Algebra</a><ul>
<li class="chapter" data-level="2.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-glimpse-ahead"><i class="fa fa-check"></i><b>2.1</b> Glimpse Ahead</a></li>
<li class="chapter" data-level="2.2" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-vectors"><i class="fa fa-check"></i><b>2.2</b> Vectors</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-physics-perspective"><i class="fa fa-check"></i><b>2.2.1</b> Physics Perspective</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-computer-science-perspective"><i class="fa fa-check"></i><b>2.2.2</b> Computer Science Perspective</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-geometric-context"><i class="fa fa-check"></i><b>2.2.3</b> Geometric Context</a></li>
<li class="chapter" data-level="2.2.4" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-coordinate-system"><i class="fa fa-check"></i><b>2.2.4</b> Coordinate System</a></li>
<li class="chapter" data-level="2.2.5" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-vector-operations"><i class="fa fa-check"></i><b>2.2.5</b> Vector Operations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-matrices"><i class="fa fa-check"></i><b>2.3</b> Matrices</a><ul>
<li class="chapter" data-level="2.3.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-special-matrices"><i class="fa fa-check"></i><b>2.3.1</b> Special Matrices</a></li>
<li class="chapter" data-level="2.3.2" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-matrix-operation"><i class="fa fa-check"></i><b>2.3.2</b> Matrix Operations</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-systems-of-equations"><i class="fa fa-check"></i><b>2.4</b> Systems Of Equations</a><ul>
<li class="chapter" data-level="2.4.1" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-matrix-vector-notation"><i class="fa fa-check"></i><b>2.4.1</b> Matrix-Vector Notation</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="intro-linalg.html"><a href="intro-linalg.html#intro-linalg-solving-systems-of-linear-equations"><i class="fa fa-check"></i><b>2.5</b> Solving Systems of Linear Equations</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quan-gen.html"><a href="quan-gen.html"><i class="fa fa-check"></i><b>3</b> Basics in Quantitative Genetics</a><ul>
<li class="chapter" data-level="3.1" data-path="quan-gen.html"><a href="quan-gen.html#single-locus-quant-trait"><i class="fa fa-check"></i><b>3.1</b> Single Locus - Quantitative Trait</a><ul>
<li class="chapter" data-level="3.1.1" data-path="quan-gen.html"><a href="quan-gen.html#qg-terminology"><i class="fa fa-check"></i><b>3.1.1</b> Terminology</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quan-gen.html"><a href="quan-gen.html#qg-frequency"><i class="fa fa-check"></i><b>3.2</b> Frequencies</a></li>
<li class="chapter" data-level="3.3" data-path="quan-gen.html"><a href="quan-gen.html#hw-eq"><i class="fa fa-check"></i><b>3.3</b> Hardy-Weinberg Equilibrium</a></li>
<li class="chapter" data-level="3.4" data-path="quan-gen.html"><a href="quan-gen.html#value-mean"><i class="fa fa-check"></i><b>3.4</b> Value and Mean</a><ul>
<li class="chapter" data-level="3.4.1" data-path="quan-gen.html"><a href="quan-gen.html#geno-value"><i class="fa fa-check"></i><b>3.4.1</b> Genotypic Values</a></li>
<li class="chapter" data-level="3.4.2" data-path="quan-gen.html"><a href="quan-gen.html#pop-mean"><i class="fa fa-check"></i><b>3.4.2</b> Population Mean</a></li>
<li class="chapter" data-level="3.4.3" data-path="quan-gen.html"><a href="quan-gen.html#breed-value"><i class="fa fa-check"></i><b>3.4.3</b> Breeding Values</a></li>
<li class="chapter" data-level="3.4.4" data-path="quan-gen.html"><a href="quan-gen.html#allele-substitution"><i class="fa fa-check"></i><b>3.4.4</b> Allele Substitution</a></li>
<li class="chapter" data-level="3.4.5" data-path="quan-gen.html"><a href="quan-gen.html#dominance-deviation"><i class="fa fa-check"></i><b>3.4.5</b> Dominance Deviation</a></li>
<li class="chapter" data-level="3.4.6" data-path="quan-gen.html"><a href="quan-gen.html#summary-of-values"><i class="fa fa-check"></i><b>3.4.6</b> Summary of Values</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="quan-gen.html"><a href="quan-gen.html#variances"><i class="fa fa-check"></i><b>3.5</b> Variances</a></li>
<li class="chapter" data-level="3.6" data-path="quan-gen.html"><a href="quan-gen.html#extension-to-more-loci"><i class="fa fa-check"></i><b>3.6</b> Extension To More Loci</a><ul>
<li class="chapter" data-level="3.6.1" data-path="quan-gen.html"><a href="quan-gen.html#epistatic-interaction"><i class="fa fa-check"></i><b>3.6.1</b> Epistatic Interaction</a></li>
<li class="chapter" data-level="3.6.2" data-path="quan-gen.html"><a href="quan-gen.html#interaction-variance"><i class="fa fa-check"></i><b>3.6.2</b> Interaction Variance</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="quan-gen.html"><a href="quan-gen.html#genetic-models"><i class="fa fa-check"></i><b>3.7</b> Genetic Models</a><ul>
<li class="chapter" data-level="3.7.1" data-path="quan-gen.html"><a href="quan-gen.html#model-usage-in-routine-evaluations"><i class="fa fa-check"></i><b>3.7.1</b> Model Usage In Routine Evaluations</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="quan-gen.html"><a href="quan-gen.html#appendix-derivations"><i class="fa fa-check"></i><b>3.8</b> Appendix: Derivations</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applied Statistical Methods in Animal Sciences</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Applied Statistical Methods in Animal Sciences</h1>
<p class="author"><em>Peter von Rohr</em></p>
<p class="date"><em>2021-02-19</em></p>
</div>
<div id="preface" class="section level1 unnumbered">
<h1>Preface</h1>
<p>This document contains the course notes for</p>
<p><strong>751-7602-00L Applied Statistical Methods in Animal Sciences</strong>.</p>
<div id="general-developments" class="section level2 unnumbered">
<h2>General Developments</h2>
<p>With the advent of <strong>Big Data</strong> (see <span class="citation">(Wikipedia <a href="#ref-Wikipedia2019" role="doc-biblioref">2019</a>)</span> and <span class="citation">(Mashey <a href="#ref-Mashey1998" role="doc-biblioref">1998</a>)</span> for a reference), it became clear that the importance of statistical methods to analyze the huge amounts of collected data would increase dramatically. Many modern statistical methods are only applicable due to the vast availability of cheap computing resources. The progress of the development that happens in the hardware manufacturing industry is often referred to by the term <strong>Moore’s Law</strong>. This law was stated as a projection as early as 1965 by one of the founders of the Intel cooperation <span class="citation">(Moore <a href="#ref-Moore1965" role="doc-biblioref">1965</a>)</span>. In a very general term, Moore’s law says that the number of circuits that could be placed on a silicon waver would double every 18 months. In a derived version the law was interpreted in a way that the performance of computers would double every 18 months. Together with the high degree of automated production of the building blocks of a computer, the prices for a single unit of computation dropped dramatically. This development made it possible that the possibility to analyze large amounts of data with modern methods can be done by almost everyone. This created very many opportunities which are actively used by many business companies. Statistical methods used to be only used by academic researchers. Nowadays almost all important decisions in business companies are done based on supporting facts that are derived from analyzing market and customer data. With that it is clear that the importance of being able to use statistical methods to analyze data is almost ubiquitous and the knowledge of these methods can be very important in many different jobs or employments.</p>
</div>
<div id="where-does-this-course-fit-in" class="section level2 unnumbered">
<h2>Where Does This Course Fit In?</h2>
<p>This course gives a short introduction to a collection of statistical methods that I believe are relevant for a wide range of topics in Animal Sciences. These methods include</p>
<ul>
<li>Multiple Linear Least Squares Regression (MLLSR)</li>
<li>Best Linear Unbiased Prediction (BLUP) which is called GBLUP when applied in the context of genomics</li>
<li>Least Absolute Shrinkage and Selection Operator (LASSO)</li>
<li>Bayesian Estimation of Unknown Parameters (BEUP)</li>
</ul>
<p>The above listed collection of statistical methods all happen to be illustrated around the same type of dataset. This dataset contains the genetic variants at many locations in the genome for a number of livestock breeding animals. Because there are many genetic locations considered in such a dataset and the locations are distributed across the complete genome, such a dataset is referred to as a <strong>genomic</strong> dataset. This type of dataset does appear in an area of livestock breeding which is called <strong>Genomic Selection</strong> (GS). GS was introduced in a seminal paper by <span class="citation">(Meuwissen, Hayes, and Goddard <a href="#ref-Meuwissen2001" role="doc-biblioref">2001</a>)</span>. This very same paper is used as a building block to explain some of the statistical methods (MLLSR and BEUP) used in this course. Furthermore the same publication illustrates that some methods (MLLSR) are not suitable for analyzing certain aspects in a genomic dataset.</p>
<p>The time available for this course is just half a semester. This leaves very little time for the introduction of each topic. As a consequence of that each topic can only be presented very superficially and students are expected to work on their own during the exercise hours. Exercises consist of sets of problems related to each topic. Problems are often to be expected to be solved using the R programming language <span class="citation">(R Core Team <a href="#ref-RCoreTeam2018" role="doc-biblioref">2018</a>)</span>.</p>
<p>This version of the course corresponds to the fifth edition. With each additional iteration of the course, improvements are sought to be implemented. Hence any input from the students are greatly appreciated.</p>
</div>
<div id="course-objectives" class="section level2 unnumbered">
<h2>Course Objectives</h2>
<p>The students are familiar with the properties of multiple linear regression and they are able to analyze simple data sets using regression methods. The students know why multiple linear regression cannot be used for problems where the number of parameters exceeds the number of observations. One such problem is the prediction of genomic breeding values used in genomic selection. The students know alternative statistical methods that can be applied in situations where the number of parameters is larger than the number of observations. Examples of such methods are BLUP-based approaches, Bayesian procedures and LASSO. The students are able to solve simple exercise problems applying BLUP-based approaches, LASSO and BEUP. The students are expected to use the statistical language and environment R <span class="citation">(R Core Team <a href="#ref-RCoreTeam2018" role="doc-biblioref">2018</a>)</span>.</p>
</div>
<div id="prerequisites" class="section level2 unnumbered">
<h2>Prerequisites</h2>
<p>Because the data that is used in this course comes from genetics, a basic level of quantitative genetics is useful for this course. All statistical models will be presented in matrix-vector notation, hence some basics of linear algebra helps in understanding the presented material. Introductory chapters to both subjects (quantitative genetics and linear algebra) are included in these course notes, but will not be discussed during the lecture. These chapters are prepared for students who feel that they need more background. But this material is left for self-studying.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Mashey1998">
<p>Mashey, John R. 1998. “Big Data ... and the Next Wave of InfraStress.” <a href="http://static.usenix.org/event/usenix99/invited%7B\_%7Dtalks/mashey.pdf">http://static.usenix.org/event/usenix99/invited{\_}talks/mashey.pdf</a>.</p>
</div>
<div id="ref-Meuwissen2001">
<p>Meuwissen, T H E, B J Hayes, and M E Goddard. 2001. “Prediction of Total Genetic Value Using Genome-Wide Dense Marker Maps.” <em>Genetics</em> 157: 1819–29.</p>
</div>
<div id="ref-Moore1965">
<p>Moore, G E. 1965. “Cramming more components onto integrated circuits.” <em>Electronics</em> 38.</p>
</div>
<div id="ref-RCoreTeam2018">
<p>R Core Team. 2018. “R: A language and environment for statistical computing.” Vienna, Austria: R Foundation for Statistical Computing. <a href="https://doi.org/R%20Foundation%20for%20Statistical%20Computing,%20Vienna,%20Austria.%20ISBN%203-900051-07-0,%20URL%20http://www.R-project.org.">https://doi.org/R Foundation for Statistical Computing, Vienna, Austria. ISBN 3-900051-07-0, URL http://www.R-project.org.</a></p>
</div>
<div id="ref-Wikipedia2019">
<p>Wikipedia, Contributors. 2019. “Big data - Wikipedia.” <a href="https://en.wikipedia.org/w/index.php?title=Big%7B\_%7Ddata%7B\&amp;%7Doldid=881938730">https://en.wikipedia.org/w/index.php?title=Big{\_}data{\&amp;}oldid=881938730</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="asm-intro.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/charlotte-ngs/gelasmss2021/edit/master/cn_asm/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-asmas-ss2021.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
