---
title: "Applied Statistical Methods In Animal Science"
author: "Peter von Rohr"
date: "22.02.2021"
output: beamer_presentation
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::knit_hooks$set(hook_convert_odg = rmdhelp::hook_convert_odg)
```

## Administration

- Course: 2 hours of lecture (2 V)
- Plan: 2 V $\rightarrow$ 1 U + 1 V (i.e., 1 hour of lecture intersperced with time to do exercises) 
- Exercises: Work on problems in R 
- Material: course notes, slides, solution to exercises
- Exam: written, date: 31.05.2021, 08:15-09:00 


## Objectives

The students 

- are familiar with the properties of __fixed linear effects models__
- are able to analyse simple data sets
- know why least squares cannot be used for genomic selection. 
- know the statistical methods used in genomic selection, such as 
    + BLUP-based approaches, 
    + Bayesian procedures and 
    + LASSO. 
- are able to solve simple exercise problems using the statistical framework R.


## Program

```{r CourseSetup, echo=FALSE, results='asis'}
### # header names of tables
vecTableHeaders <- c("Week", "Date", "Topic")
# define course start date
dCourseStart <- as.Date("2021/02/22")
# set number of weeks in semester
nNrSemesterWeeks <- 15
# define columns for weaks, dates, and subjects
Week <- 1:nNrSemesterWeeks
Date <- format(seq(dCourseStart, by = "week", length.out = nNrSemesterWeeks), "%d.%m")
Topic <- vector(mode = "character", length = nNrSemesterWeeks)
# subjects per week
Topic[1] <- "Introduction to Applied Statistical Methods"
Topic[2] <- "Linear Fixed Effect Models"
Topic[3] <- "GBLUP - Marker-Effects Models"
Topic[4] <- "GBLUP - Breeding Value Models"
Topic[5] <- "Lasso"
Topic[6] <- "Bayesian Approaches"
Topic[7] <- "__Easter Monday__"
Topic[8] <- "Introduction to Genetic Evaluation of Livestock"
Topic[9] <- "Model Selection"
Topic[10] <- "Variance Components"
Topic[11] <- "Genetic Groups and Longitudinal Data"
Topic[12] <- "Genomic Selection"
Topic[13] <- "Questions, Test Exam"
Topic[14] <- "__Pfingstmontag__"
Topic[15] <- "Exams" 

dfCnTable <- data.frame(Week, Date, Topic, stringsAsFactors = FALSE)
colnames(dfCnTable) <- vecTableHeaders
knitr::kable(dfCnTable)
```


## Information

- Website: https://charlotte-ngs.github.io/gelasmss2021/
- Topics for master thesis: https://charlotte-ngs.github.io/gelasmss2021/misc/MasterThesisTopics_SS2021.html
- Exam: 31.05.2021 08:15 -- 09:00


## This Course

- Use dataset that is used to predict genomic breeding values and introduce four methods

1. Fixed Linear Effects Models - Least Squares
2. GBLUP - genomic version of BLUP
3. LASSO - still fixed linear effects model, but modified parameter estimation
4. Bayesian approach to estimate unknown parameter


## Significance

- Why is this important? 
- Is this only relevant for animal breeding? 
- What about the rest of animal science?

- General trend of collecting data has led to development of `Big Data`
- Examples
    + Presidential campains in the US
    + Health care
    + Face recognition
    + Agriculture: Smart Farming
    + Animal Science


## Face Recognition

```{r face-recognition, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='12cm'}
#rmdhelp::use_odg_graphic(ps_path = 'odg/face-recognition.odg')
knitr::include_graphics(path = "odg/face-recognition.png")
```


## Traditional Animal Breeding

- Before 2006
- Data collected for other purposes were used to predict breeding values
- Predicted breeding values as side-product


## Data Logistics

```{r datalogistics, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/datalogistics.odg")
knitr::include_graphics(path = "odg/datalogistics.png")
```

 
## Genomic Selection

- Same goal as in traditional breeding: Find animals with best genetic potential as parents of next generation
- New: use additional source of information
- __Genomic__ information
    + spread accross whole genome
    + single nucleotide polymorphisms (SNP)
- Introduction: 

\small "> Meuwissen THE, Hayes BJ, Goddard ME (2001) Prediction of total genetic value
using genome-wide dense marker maps. Genetics 157:1819–1829"

- Popularisation:

\small "> L. R. Schaeffer. Strategy for applying genome-wide selection in dairy cat- tle. Journal of Animal Breeding and Genetics, 123(4):218–223, 2006. ISSN 09312668. doi: 10.1111/j.1439-0388.2006.00595.x."


## SNP

```{r snparchitecture, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/snparchitecture.odg")
knitr::include_graphics(path = "odg/snparchitecture.png")
```
 

## QTL

```{r linkagesnpqtl, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/linkagesnpqtl.odg")
knitr::include_graphics(path = "odg/linkagesnpqtl.png")
```
 

## Linkage

- Flanking SNPs and QTL not independent passed on from parents to progeny
- Favorable QTL-allele linked with a given SNP-allele
- QTL is unknown, but use SNPs close to QTL as information for selection


## Monogenic Model

- Assume quantitative trait is influenced by one locus only
- Locus is bi-allelic $\rightarrow$ two alleles ($G_1$ and $G_2$) and three genotypes
- Look at Distribution of trait values for three different genotypes


## Distribution No Effect

```{r monogenicmodeltraitdistributionnoeffect, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/monogenicmodeltraitdistributionnoeffect.odg")
knitr::include_graphics(path = "odg/monogenicmodeltraitdistributionnoeffect.png")
```


## Distribution With Effect

```{r monogenicmodeltraitdistribution, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/monogenicmodeltraitdistribution.odg")
knitr::include_graphics(path = "odg/monogenicmodeltraitdistribution.png")
```
  

## Breeding Value

- Definition: Two times deviation from large number of offspring from population mean
- Assume: Hardy-Weinberg equilibrium
- Compute population mean as expected value of genotypic values
- Compute expected genotypic value of offspring for each of the three parental genotypes
- Assume purely additive loci, hence $d = 0$


## Genomic Breeding Value

- Take into account many loci
- Approximate unknown QTL with linked SNP
- Estimate $a$-effects from monogenic model
- Compute genomic breeding values for all loci based on $a$ effects


## Two Approaches

1. Two Step Procedure (used currently in Swiss Dairy Cattle)
2. Single Step 


## Two Step

```{r twostepprocedure, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/twostepprocedure.odg")
knitr::include_graphics(path = "odg/twostepprocedure.png")
```


## Single Step

- Combine all information into one single BLUP-based analysis
- Problem: Determine covariance between animals with and without genomic information


## Summary: Traditional versus Genomic Selection

```{r summarytraditionalvsgs, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", out.width='11cm'}
#rmddochelper::use_odg_graphic(ps_path = "odg/summarytraditionalvsgs.odg")
knitr::include_graphics(path = "odg/summarytraditionalvsgs.png")
```


  
